src/plugins/hyperbee/hyperbee.cpp

```diff
@@ -0,0 +1,1033 @@
+ * student of FAU Erlangen-Nürnberg (DE) in cooperation with Politecnico   *
+ * di Milano (IT) during 2020/2021.                                        *
+ * The main goal of this plugin is to efficiently dump executable pages    *
+ * which afterwards could be processed by analysis tools to detect         *
+ * malware. The goal is reached by installing several traps:               *
+ * 1. mmAccessFaultTrap: Whenever a MmAccessFault is called (to commit     *
+ *    virtual memory) the regarding virtual address is saved. Additionally *
+ *    the second trap is set up:                                           *
+ * 2. When MmAccessFault returns to the original code it provides the      *
+ *    physical address the virtual address was assigned to. This           *
+ *    information is saved as well and used to set up a third trap:        *
+ * 3. Whenever instructions are fetched from the physical address (frame)  *
+ *    to be executed, this trap dumps the memory of the frame or even the  *
+ *    whole vad node.                                                      *
+ *    To be aware of further executions of this page but prevent duplicate *
+ *    dumps, this execution trap is replaced by a fourth, write trap.      *
+ * 4. The write trap is reacting on any writes at the monitored memory and *
+ *    replaces itself with an executable trap (3). By continuously         *
+ *    swapping trap 3 and 4 memory a dump of the same memory area is only  *
+ *    performed, if the memory was modified in the mean time.              *
+ **************************************************************************/
+
+#include <glib.h>
+#include <cinttypes>
+#include <sys/stat.h>
+#include <libvmi/libvmi.h>
+#include <string>
+#include <filesystem>
+#include <vector>
+#include <cstdio>
+#include <cstring>
+
+#include <libdrakvuf/json-util.h>
+#include "hyperbee.h"
+#include "plugins/output_format.h"
+#include "private.h"
+
+/**
+ * Saves a pointer to the plugin and the starting virtual address of the frame.
+ * This struct is passed between the different traps.
+ */
+struct fault_data {
+    hyperbee *plugin;
+    addr_t pageVA;
+};
+
+/**
+ * This is the callback of the execute trap.
+ * @param drakvuf the dravuf plugin
+ * @param info information regarding the trap event
+ * @return VMI_EVENT_RESPONSE_NONE, but this return information is currently not used by drakvuf.
+ */
+static event_response_t execute_faulted_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info);
+
+/**
+ * Creates an execute trap for the given (guest) frame number and passes the given fault data
+ * @param gfn the (guest) frame number. If some executable instructions are fetched from this frame, the trap gets active
+ * @param faultData information which is required for logging. It is passed between the traps.
+ * @return the execute trap that needs to be added to drakvuf
+ */
+drakvuf_trap_t *createExecuteTrap(addr_t gfn, fault_data *faultData);
```

Comment:
I think there is a mixed naming convention within this file, maybe it should be `create_execute_trap`?

Reply:
Thanks for the hint. Are there any "common" C++ or drakvuf naming conventions? I'll have a look at it!

Reply:
Do you want me to refactor this as well to libdrakvuf? I'm not sure if you want me to do that since it somehow "looks" like it belongs to your upper comment?

Reply:
No, this doesn't belong to the general ("upper") comment.

Reply:
Well, actually there is an idea to use google style in C++ code, which uses camelCase :P

Reply:
> Are there any "common" C++ or drakvuf naming conventions?

I'm not sure if it is written anywhere, but I think in DRAKVUF we are using `this_kind_of_case` (almost) everywhere. For me it is sufficient that the naming is consistent accross the module.

Reply:
Ok, I have no problems with changing the overall code style and I don't mind if it the Google style or sth else.
With the last commit I made the function naming consistent.